subprojects/core/src/integTest/groovy/org/gradle/api/internal/buildevents/BuildResultLoggerFunctionalSpec.groovy

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.buildevents
+
+import org.fusesource.jansi.Ansi
+import org.gradle.integtests.fixtures.AbstractConsoleFunctionalSpec
+
+class BuildResultLoggerFunctionalSpec extends AbstractConsoleFunctionalSpec {
+
+    def setup() {
+        executer.withStackTraceChecksDisabled()
+    }
+
+    def "Failure status is logged even in --quiet"() {
+        given:
+        buildFile << "task fail { doFirst { assert false } }"
+
+        when:
+        executer.withQuietLogging()
+        fails('fail')
+
+        then:
+        result.output.contains("BUILD FAILED")
+    }
+
+    def "Failure message is logged bold and red"() {
```

Comment:
We should apply this to all log levels. I think we should also cover a bit more stuff, to pay down some testing debt. For example, that this message plus the build summary appear as the last 2 lines of output.